msmart/device/AC/command.py

```diff
@@ -114,9 +114,23 @@ def payload(self) -> bytes:
         beep = 0x40 if self.beep_on else 0
         power = 0x1 if self.power_on else 0
 
-        # Build target temp and mode byte
-        fractional, integral = math.modf(self.target_temperature)
-        temperature = (int(integral) & 0xF) | (0x10 if fractional > 0 else 0)
+        # Get integer and fraction components of target temp
+        fractional_temp, integral_temp = math.modf(self.target_temperature)
+        integral_temp = int(integral_temp)
+
+        if 17 <= integral_temp <= 30:

```

Comment:
The reference code here is inconsistent. 

The Lua checks for >= 16 when setting temperature1 and the checks for < 17 or > 30 for temperature2.

The plugin checks for >= 16 and <= 30 when setting temp1 and nothing for temp2